tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -663,3 +797,131 @@ func disableQinQOnSwitch(switchCredentials *sriovenv.SwitchCredentials, switchIn
+
+	return []string{"/bin/bash", "-c", baseCmd}
+}
+
+func defineTestClientPmdCmd(ethPeer, pciAddress string) []string {
+	baseCmd := fmt.Sprintf(
+		"timeout -s SIGKILL 20 dpdk-testpmd --vdev=virtio_user0,path=/dev/vhost-net,iface=net2 -a %s -- "+
+			"--forward-mode txonly --eth-peer=0,%s --stats-period 5", pciAddress, ethPeer)
+
+	return []string{"/bin/bash", "-c", baseCmd}
+}
+
+func defineAndCreateServerDPDKPod(
+	podName,
+	nodeName string,
+	serverPodNetConfig []*multus.NetworkSelectionElement,
+	podCmd []string) *pod.Builder {
+	var rootUser int64
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"},
+		},
+	}
+
+	dpdkContainer := pod.NewContainerBuilder(podName, NetConfig.DpdkTestContainer, podCmd)
+	dpdkContainerCfg, err := dpdkContainer.WithSecurityContext(&securityContext).
+		WithResourceLimit("2Gi", "1Gi", 4).
+		WithResourceRequest("2Gi", "1Gi", 4).WithEnvVar("RUN_TYPE", "testcmd").
+		GetContainerCfg()
+
+	Expect(err).ToNot(HaveOccurred(), "Fail to set dpdk container")
+
+	dpdkPod := pod.NewBuilder(APIClient, podName, tsparams.TestNamespaceName,
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+	dpdkPort0Cmd := `port stop 0
+tx_vlan set 0 100
+port start 0
+start
+`
+	configMapData := map[string]string{"cmd_file": dpdkPort0Cmd}
+	configMap, err := configmap.NewBuilder(APIClient, "dpdk-port-cmd", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+	dpdkPod.WithLocalVolume(configMap.Definition.Name, "/etc/cmd")
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(4 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Fail to create server pod")
+
+	return dpdkPod
+}
+
+func defineAndCreateClientDPDKPod(
+	podName,
+	nodeName string,
+	serverPodNetConfig []*multus.NetworkSelectionElement) *pod.Builder {
+	var rootUser int64
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"},
+		},
+	}
+
+	dpdkContainerCfg, err := pod.NewContainerBuilder(podName, NetConfig.DpdkTestContainer,
+		[]string{"/bin/bash", "-c", "sleep INF"}).WithSecurityContext(&securityContext).
+		WithResourceLimit("2Gi", "1Gi", 4).
+		WithResourceRequest("2Gi", "1Gi", 4).WithEnvVar("RUN_TYPE", "testcmd").GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Fail to set dpdk container")
+
+	dpdkPod := pod.NewBuilder(APIClient, podName, tsparams.TestNamespaceName,
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(10 * time.Minute)
```

Comment:
Please make the same as for the server pod

Reply:
changed